### PR TITLE
added mypy.ini

### PR DESCRIPTION
**Pull request by KerstenBreuer**
Moved all mypy configs from Github Actions and Pre-Commit to a mypy.ini

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Global options:
+
+[mypy]
+disable_error_code = import
+show_error_codes = True
+exclude = (?x)(
+    build/lib/
+  )
```
Comment (Cito) [inline mypy.ini:8]: Some more options that can be helpful:
```
warn_redundant_casts = True
warn_unused_ignores = True
```
As mypy improved, this helped me to find which casts and ignores became unnecessary, and could be removed.
```
check_untyped_defs = True
```
To ensure that all functions are annotated.

Comment (Cito) [reply mypy.ini:8]: We could also name this `.mypy.ini` since most of the other config files also are hidden with a dot. mypy recognizes both.

**Review by Cito — APPROVED** [review]

File: mypy.ini (hunk 1 above)
Comment (KerstenBreuer) [reply mypy.ini]: Thanks a lot for the suggestions, implemented them.

File: scripts/license_checker.py [hunk 2]
```diff
@@ -61,6 +61,7 @@
     "requirements.txt",
     ".vscode",
     ".mypy_cache",
+    "mypy.ini",
```
Comment (Cito) [inline scripts/license_checker.py:64]:
```suggestion
    ".mypy.ini",
```
Since this was renamed.

**Pull request merged**